src/backend/distributed/commands/multi_copy.c

```diff
@@ -566,6 +562,7 @@ CopyToNewShards(CopyStmt *copyStatement, char *completionTag, Oid relationId)
 	errorCallback.callback = CopyFromErrorCallback;
 	errorCallback.arg = (void *) copyState;
 	errorCallback.previous = error_context_stack;
+	error_context_stack = &errorCallback;
```

Comment:
Actually, this is also done below

Reply:
Yeah you are right, actually `CopyToExistingShards` seems to be missing some lines. `CopyToNewShards` and `CopyToExistingShards` had some inconsistency. I have updated it, can you review the change again @marcocitus?